src/query/interpret/eval.hpp

```diff
@@ -479,9 +479,37 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
       case TypedValue::Type::Null:
         return TypedValue(ctx_->memory);
       case TypedValue::Type::Vertex:
-        return TypedValue(GetProperty(expression_result_ptr->ValueVertex(), property_lookup.property_), ctx_->memory);
+        if (property_lookup.evaluation_mode_ == PropertyLookup::EvaluationMode::GET_ALL_PROPERTIES) {
+          auto symbol_pos = static_cast<Identifier *>(property_lookup.expression_)->symbol_pos_;
+          if (!ctx_->property_lookups_cache.contains(symbol_pos)) {
+            ctx_->property_lookups_cache[symbol_pos] = GetAllProperties(expression_result_ptr->ValueVertex());
+          }
+
+          auto property_id = ctx_->properties[property_lookup.property_.ix];
+          if (!ctx_->property_lookups_cache.at(symbol_pos).contains(property_id)) {
+            return TypedValue(ctx_->memory);
```

Comment:
If this threw an exception, it would clash with the Cypher specification.

> Conceptually, `null` means 'a missing unknown value' and it is treated somewhat differently from other values. For example getting a property from a node that does not have said property produces `null`.

[Cypher Query Language Reference (Version 9)](https://s3.amazonaws.com/artifacts.opencypher.org/openCypher9.pdf), page 22